test/hotspot/jtreg/containers/cgroup/NestedCgroup.java

```diff
@@ -0,0 +1,268 @@
+                    "", "",
+                    "cgdelete: cannot remove group '" + CGROUP_OUTER + "': No such file or directory");
+            } catch (IOException e) {
+                if (e.toString().equals("java.io.IOException: Cannot run program \"cgdelete\": error=2, No such file or directory")) {
+                    throw new SkippedException("libcgroup-tools is not installed");
+                }
+                throw e;
+            }
+
+            // Alpine Linux 3.20.1 needs cgcreate1 otherwise:
+            // cgcreate: can't create cgroup CONTROLLERS_PATH_INNER: No such file or directory
+            List<String> cgcreate1 = new ArrayList<>();
+            cgcreate1.add("cgcreate");
+            cgcreate1.add("-g");
+            cgcreate1.add(CONTROLLERS_PATH_OUTER);
+            pSystem(cgcreate1,
+                "cgcreate: can't create cgroup " + CGROUP_OUTER + ": Cgroup, operation not allowed", "Missing root permission",
+                "cgcreate: can't create cgroup " + CGROUP_OUTER + ": Cgroup, requested group parameter does not exist", "Missing root permission",
+                "");
+
+            List<String> cgcreate2 = new ArrayList<>();
+            cgcreate2.add("cgcreate");
+            cgcreate2.add("-g");
+            cgcreate2.add(CONTROLLERS_PATH_INNER);
+            pSystem(cgcreate2, "", "", "", "", "");
+
+            String provider = Metrics.systemMetrics().getProvider();
+            System.err.println("Metrics.systemMetrics().getProvider() = " + provider);
+            boolean isCgroup2;
+            if ("cgroupv1".equals(provider)) {
+              isCgroup2 = false;
+            } else if ("cgroupv2".equals(provider)) {
+              isCgroup2 = true;
+            } else {
+              throw new IllegalArgumentException();
+            }
+            System.err.println("isCgroup2 = " + isCgroup2);
+
+            String mountInfo;
+            try {
+                mountInfo = Files.readString(Path.of(MOUNTINFO));
+            } catch (NoSuchFileException e) {
+                throw new SkippedException("Cannot open " + MOUNTINFO);
+            }
+
+            Matcher matcher = Pattern.compile("^(?:\\S+\\s+){4}(\\S+)\\s.*\\scgroup(?:2(?:\\s+\\S+){2}|\\s+\\S+\\s+(?:\\S*,)?memory(?:,\\S*)?)$", Pattern.MULTILINE).matcher(mountInfo);
+            if (!matcher.find()) {
+                System.err.println(mountInfo);
+                throw new SkippedException("cgroup/cgroup2 filesystem mount point not found");
+            }
+            sysFsCgroup = matcher.group(1);
+            System.err.println("sysFsCgroup = " + sysFsCgroup);
+
+            System.err.println(LINE_DELIM + " " + (isCgroup2 ? "cgroup2" : "cgroup1") + " mount point: " + sysFsCgroup);
+            memory_max_filename = isCgroup2 ? "memory.max" : "memory.limit_in_bytes";
+            Files.writeString(Path.of(sysFsCgroup + "/" + CGROUP_OUTER + "/" + memory_max_filename), "" + MEMORY_MAX_OUTER);
```

Comment:
This still doesn't work on cgv1 (hybrid). The reg-ex pattern matching is wrong. I'd suggest to simplify this by using `cgget` and `cgset` with forked processes. Something like this (depending on the version use `memory.max` or `memory.limit_in_bytes`):

```
$ cgcreate -g memory:/jdktest128331
$ cgcreate -g memory:/jdktest128331/inner
$ cgset -r memory.limit_in_bytes=512000 /jdktest128331/inner
$ cgget -n -v -r memory.limit_in_bytes /jdktest128331/inner
512000
```